src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -164,7 +170,11 @@ def run_python_test(transitive_hydrated_target, pytest, python_setup, source_roo
 
   request = ExecuteProcessRequest(
     argv=(python_binary, './{}'.format(output_pytest_requirements_pex_filename)),
-    env={'PATH': os.pathsep.join(python_setup.interpreter_search_paths)},
+    env={
+      'PATH': text_type(os.pathsep.join(python_setup.interpreter_search_paths)),
+      'LANG': text_type(os.environ.get('LANG', '')),
+      'LC_ALL': text_type(os.environ.get('LC_ALL', '')),
```

Comment:
What are these necessary for? Would it be useful for other rules? Consider adding a helper method, an intermediate ruleset (to generate an `ExecuteProcessRequest`), or a TODO here to avoid sequestering this knowledge and risking bitrot (maybe).

Reply:
They're necessary to ensure that Python uses UTF-8 whenever it does `with open()`. Otherwise, on my system it's using ASCII codec :/

Will somehow move these into the subsystem we've been talking about. Haven't taken a look at how to do so yet.